types/src/v0/impls/l1.rs

```diff
@@ -77,6 +78,27 @@ impl L1BlockInfo {
     }
 }
 
+impl RpcClient {
+    fn http(url: Url) -> Self {
+        Self::Http(Http::new(url))
+    }
+
+    async fn ws(url: Url, retry_delay: Duration) -> anyhow::Result<Self> {
+        Ok(Self::Ws {
+            conn: Arc::new(RwLock::new(Ws::connect(url.clone()).await?)),
+            reconnect: Default::default(),
+            retry_delay,
+            url,
+        })
+    }
+
+    async fn stop(&self) {
```

Comment:
Yeah it stops all the background tasks (the reconnect task and the snapshot updating task). Ideally this would take `self` by value and just destroy the whole client but it's not easy to implement like that because L1 client is behind an `Arc` in `NodeState`.

Maybe like `shut_down_tasks`, and also rename `start` to `spawn_tasks`?

Reply:
Sounds good to me